File: Library/core/src/main/java/com/emedinaa/core/ui/dialog/SimpleDialog.java
package com.emedinaa.core.ui.dialog;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

/**
 * @author : Eduardo Medina
 * @see : https://developer.android.com/index.html
 * @since : 8/18/18
 */
public class SimpleDialog extends DialogFragment {

    private SimpleDialogListener listener;
    private String title="";
    private String message="";

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
     if (context instanceof SimpleDialogListener) {
         listener = (SimpleDialogListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement SimpleDialogListener");
        }
    }
    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(getArguments()!=null){
            title=getArguments().getString("TITLE", "");
            message=getArguments().getString("MESSAGE", "");
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // Use the Builder class for convenient dialog construction
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(title)
                .setMessage(message)
                .setPositiveButton("Aceptar", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        if(listener!=null){
                            listener.onAction();
                        }
                    }
                })
                .setOnCancelListener(new DialogInterface.OnCancelListener() {
                    @Override
                    public void onCancel(DialogInterface dialogInterface) {
                        if(listener!=null){
                            listener.onAction();
                        }
                    }
                });
                /*.setNegativeButton("Cancelar", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        // User cancelled the dialog
                    }
                });*/
        // Create the AlertDialog object and return it
        return builder.create();
    }

    public interface SimpleDialogListener{
        void onAction();
    }
}

File: KotlinL4/app/src/main/java/com/emedinaa/kotlinapp/dialog/SimpleDialog.kt
package com.emedinaa.kotlinapp.dialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.emedinaa.kotlinapp.R
import android.util.Log


class SimpleDialog:DialogFragment() {

    private lateinit var mContext:Context

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(mContext)

        builder.setTitle("Aviso !")
            .setMessage(R.string.dialog_fire_missiles)
            .setPositiveButton(R.string.fire) { _, _ ->
                // FIRE ZE MISSILES!
                log("Dialog aceptar")
            }
            .setNegativeButton(R.string.scancel){ _, _ ->
                // FIRE ZE MISSILES!
                log("Dialog cancelar")
            }
        return builder.create()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mContext= context
    }

    private fun log(message: String) {
        Log.d("CONSOLE", message)
    }
}